### PR TITLE
Removed MainReactNativeHost from samples and use ReactApplication properties instead

### PR DESCRIPTION
**Pull request by vmoroz**
Requiring the MainReactNativeHost component in each sample project that we plan to generate by CLI seems to be an unnecessary overhead. All application-wide properties can be set directly on the ReactApplication component.

In this change:
- we removed the MainReactNativeHost from samples;
- added new properties to ReactApplication to be specified in constructor instead of MainReactNativeHost.

Note, that the new ReactApplication properties are the same as they used to be in MainReactNativeHost. We just let them to be set in constructor instead of overriding. In one of the next changes we might change the property names and how they are grouped.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3501)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/ReactApplication.h [hunk 1]
```diff
@@ -1,48 +1,123 @@
...
   virtual void OnNavigationFailed(
       IInspectable const &,
       NavigationFailedEventArgs const &);
 
  protected:
-  virtual winrt::Microsoft::ReactNative::
-      ReactApplicationDelegate __stdcall CreateReactApplicationDelegate() {
-    return winrt::Microsoft::ReactNative::ReactApplicationDelegate(*this);
-  }
+  virtual ReactApplicationDelegate __stdcall CreateReactApplicationDelegate();
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactApplication.h]: >__stdcall [](start = 35, length = 9)

why is this necessary?

Comment (vmoroz) [reply vnext/Microsoft.ReactNative/ReactApplication.h]: I am not sure. It was there before. I would like to remove the ReactApplicationDelegate in future iterations. IMHO, there are too many types in the API that we must remove.
